### ISSUE TITLE
Upgrades: cause boot failures [20190119]

### ISSUE THREAD

**Issue by mmarmm** [aarch64-laptops/prebuilt#10]
Running, " sudo apt dist-upgrade " might upgrade the kernel and/or grub prevent the device from booting again without intervention.

**Comment by jhugo** [comment]
Understood.  Other than artificially versioning the installed GRUB/kernel so that it never upgrades, do you have a suggestion for how to manage this?

**Comment by mmarmm** [comment]
If this report is accurate, we should pull in whatever patches that allow the Debian Buster installer to boot and we don't have to worry about it ever again: https://github.com/aarch64-laptops/prebuilt/issues/15#issuecomment-459927322

**Comment by mmarmm** [comment]
@aarch64-laptops As you pointed out, apt-mark hold will work to prevent soft bricking; I tried, " sudo apt-hold grub* ", which is overkill but it works. I even tried a release upgrade and that also worked but only bionic -> cosmic, when I tried cosmic -> disco it complained about the hold. I think only grub-efi-arm64* needs to be held. I'm looking into the other options and pinning to see but a PPA might fix that last upgrade problem I hit.

**Comment by aarch64-laptops** [comment]
No problem.  If you want to knock something up, perhaps we can work together to upload something to:

https://launchpad.net/~aarch64-laptops

**Comment by RussianNeuroMancer** [comment]
Seems like patch will be merged to Ubuntu soon: https://bugs.launchpad.net/ubuntu/+source/grub2/+bug/1812317/comments/5 (at least I hope so).

**Comment by aarch64-laptops** [comment]
> Seems like patch will be merged to Ubuntu soon: https://bugs.launchpad.net/ubuntu/+source/grub2/+bug/1812317/comments/5 (at least I hope so).

Right, but it will have to be SRUed in order to make it into existing releases.

We'll push for this to happen and keep you updated.

**Comment by lag-linaro** [comment]
We now have Grub and Kernel PPAs, meaning that upgrading all Bionic packages is now possible.

The trouble comes when upgrading to Cosmic, as boot culminates with a flashing cursor:

```
gnome-shell[1817]: Failed to create backend: Could not find a primary drm kms device
gnome-session[1802]: gnome-session-binary[1802]: WARNING: App 'org.gnome.Shell.desktop' exited with code 1
gnome-session-binary[1802]: WARNING: App 'org.gnome.Shell.desktop' exited with code 1
gnome-session-binary[1802]: Unrecoverable failure in required component org.gnome.Shell.desktop
```

Not sure why this worked perfectly on Bionic, but fails on Cosmic.
